pkg/inject/report.go

```diff
@@ -47,7 +60,7 @@ func (r Report) Injectable() bool {
 
 // update updates the report for the provided resource conf.
 func (r *Report) update(conf *ResourceConfig) {
-	r.InjectDisabled = conf.pod.Meta.GetAnnotations()[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled
+	r.InjectDisabled = conf.pod.meta.GetAnnotations()[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled || conf.nsAnnotations[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled
```

Comment:
So this means that if a namespace is tagged with `inject:disabled` then any pod injection will be disabled even if they have `inject:enabled` explicitly set?
We could think the annotation at the namespace level being set by an admin with higher access, who would want to explicitly enable/disable the possibility of injection by less-privilege users, in which case this behavior would make sense.
OTOH, we could also think that we'd like to leave the door open for more flexibility by being able to override everything at the pod-level, including the ability to inject.
So I don't know which way we should go...

Reply:
If pod has `inject: enabled`, it will take precedence over any namespace level annotation. See [this test](https://github.com/linkerd/linkerd2/blob/b00890144920a2166225785c3f53b5bf67e57671/pkg/inject/report_test.go#L91-L100), for `report.Injectable()`. Let me know if you think there is a :beetle:  here.

Reply:
You meant [this test](https://github.com/linkerd/linkerd2/blob/b00890144920a2166225785c3f53b5bf67e57671/pkg/inject/report_test.go#L78-L89) instead, right?
It appears that `nsAnnotations` is not being used in that test suite :eyes: